### PR TITLE
fix "Save" in "Unsaved changes" doesn't do anything

### PR DESCRIPTION
**Pull request by giemic8**
# Description

### Contribution Checklist:

- [X] **The pull request only addresses one issue or adds one feature.**
- [X] **The pull request does not introduce any breaking changes**
- [X] **I have added screenshots or gifs to help explain the change if applicable.**
- [X] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [X] **Create an issue and link to the pull request.**

Note: Keeping the PR small and focused helps make it easier to review and merge. If you have multiple changes you want to make, please consider submitting them as separate pull requests.

### Publishing to New Package Managers

Please see [here](../publishing.md) for more information.

### CODE REVIEW

**Comment by giemic8** [comment]
#1491

File: packages/bruno-app/src/utils/collections/index.js [hunk 1]
```diff
@@ -417,7 +417,9 @@ export const transformRequestToSaveToFilesystem = (item) => {
   });
 
   if (itemToSave.request.body.mode === 'json') {
-    itemToSave.request.body.json = replaceTabsWithSpaces(itemToSave.request.body.json);
+    const copyItemToSave = JSON.parse(itemToSave);
```
Comment (rsilvr) [inline packages/bruno-app/src/utils/collections/index.js]: As this bug originates from the fact that object coming from redux store are [not extensible](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Cant_define_property_object_not_extensible), cloning the items sent to `saveMultipleRequests` action you could avoid this error if a similar mutation is done elsewhere.
You can see the same approach in `saveRequest` action:
```js
const collectionCopy = cloneDeep(collection);
const item = findItemInCollection(collectionCopy, itemUid);
```

Comment (giemic8) [reply packages/bruno-app/src/utils/collections/index.js]: II have changed this

File: packages/bruno-app/src/utils/collections/index.js [hunk 2]
```diff
@@ -417,7 +417,9 @@ export const transformRequestToSaveToFilesystem = (item) => {
   });
 
   if (itemToSave.request.body.mode === 'json') {
-    itemToSave.request.body.json = replaceTabsWithSpaces(itemToSave.request.body.json);
+    const copyItemToSave = cloneDeep(itemToSave);
```
Comment (rsilvr) [inline packages/bruno-app/src/utils/collections/index.js:420]: Actually, in my last comment I meant cloning the items that are received in the `saveMultipleRequests` function. Although doing the change here also solves the bug, I believe that applying this fix there can avoid the bug arising from a similar mutation in another function or even in another part of `transformRequestToSaveToFilesystem`.

Comment (giemic8) [reply packages/bruno-app/src/utils/collections/index.js:420]: Should I do anything or can I close this PR?

Comment (rsilvr) [reply packages/bruno-app/src/utils/collections/index.js:420]: I've made this comment as a suggestion of improvement, but I don't actually have the necessary permissions to approve it and merge it. For that we rely in @helloanoop

**Pull request closed without merging**